subprojects/core/src/main/java/org/gradle/process/internal/DefaultJavaForkOptions.java

```diff
@@ -46,15 +49,29 @@ public DefaultJavaForkOptions(FileResolver resolver, Jvm jvm) {
     }
 
     public List<String> getAllJvmArgs() {
-        return options.getAllJvmArgs();
+        if (jvmArgumentProviders != null) {
```

Comment:
Skip empty, too? Maybe introduce `hasArgumentProviders()`?

Reply:
Added the method and now checking empty, too.